react/testing.md

```diff
@@ -276,16 +305,262 @@ For components that are basically a composite of other components ([organisms](h
+
+For HTML elements and their attributes, focus on the markup that's displayed based on logic. Static markup _can_ be tested, but has very diminishing returns.
+
+```js
+// good
+it('includes the disabled CSS class when `isDisabled` is `true`', () => {
+    let wrapper = mount(<Spinner isDisabled={true} />);
+
+    // assert that the Spinner's container/root node has
+    // "spinner--disabled" CSS class added
+    expect(wrapper).to.have.className('spinner--disabled');
+});
+
+// bad (also asserts that the container node is a <div>)
+it('includes the disabled CSS class when `isDisabled` is `true`', () => {
+    let wrapper = mount(<Spinner isDisabled={true} />);
+
+    // assert that the Spinner's container/root node
+    // is a <div>
+    expect(wrapper).to.have.tagName('div');
+
+    // assert that the Spinner's container/root node has
+    // "spinner--disabled" CSS class added
+    expect(wrapper).to.have.className('spinner--disabled');
+});
+```
+
+There's nothing fundamentally wrong with testing that the container/root node is a `<div>` in the "bad" example, but it os unnecessary since it's static and never changes. In fact, if your test file was filled with these sorts of additional assertions, it could make your test cases more fragile because presentational changes in your component could cause your component's tests to fail.
+
 **[⬆ back to top](#table-of-contents)**
 
 ## Testing events
 
+As mentioned in our [Testing philosophy](#testing-philosophy), part of the output of your component are the callback handlers it invokes. These event callbacks are functions passed as props to your component and need to be tested.
+
+Test event callbacks by triggering the events that in turn will invoke the callback handler. The type of event triggered depends on whether the component contains HTML markup or child components.
+
+### Testing events triggered by DOM
+
+If you are testing an event callback that is triggered by a DOM event (such as `onClick` of an `<button>` node), you will need to simulate that DOM event. You will also need to stub the event callback prop to assert that it is being called with the correct arguments.
+
+Let's say that there is a `TextInput` component that wraps an `<input type="text" />` DOM node. The `TextInput` has an `onChange` prop that gets called whenever the input field value changes. The `onChange` prop is also called with the current value that's in the input field. The test case would be set up like:
+
+```js
+it('properly fires `onChange` when input changes', () => {
+    let stubbedOnChange = sinon.stub();
```

Comment:
Yeah I agree, you could add a comment to remember to `import / require` sinon.
It is not really a big issue.